src/plugins/tscn/tscnplugin.cpp

```diff
@@ -0,0 +1,666 @@
+                        .arg(QString::number(x1), QString::number(y1), QString::number(x2), QString::number(y2))
+                        .toUtf8());
+
+                    break;
+                }
+                case MapObject::Polygon: {
+                    auto polygon = object->polygon().toPolygon();
+                    bool first = true;
+                    for (auto point : polygon) {
+                        if (!first)
+                            device->write(", ");
+                        auto x = point.x() - centerX;
+                        auto y = point.y() - centerY;
+                        flipState(x, y, flippedState);
+                        device->write(QString::number(x).toUtf8());
+                        device->write(", ");
+                        device->write(QString::number(y).toUtf8());
+                        first = false;
+                    }
+                    break;
+                }
+                default:
+                    throw tscnError("Godot exporter only supports collisions that are rectangles or polygons.");
+            }
+
+            device->write(")\n");
+            polygonId++;
+        }
+    }
+
+    return foundCollisions;
+}
+
+// Write the tileset
+// If you're creating a reusable tileset file, pass in a new file device and
+// set isExternal to true, otherwise, reuse the device from the tscn file.
+static void writeTileset(const Map *map, QFileDevice* device, bool isExternal, AssetInfo& assetInfo) {
+    bool foundCollisions = false;
+
+    // One Texture2D and one TileSetAtlasSource per tileset, plus a resource node
+    auto loadSteps = assetInfo.tilesetInfo.size() * 2 + 1;
+
+    if (isExternal) {
+        device->write(QString("[gd_resource type=\"TileSet\" load_steps=%1 format=3]\n\n")
+            .arg(QString::number(loadSteps)).toUtf8());
+    }
+
+    // Texture2D nodes
+    for (auto it = assetInfo.tilesetInfo.begin(); it != assetInfo.tilesetInfo.end(); ++it) {
+        if (it->usedTiles.size() == 0)
+            continue;
+
+        device->write(QString("[ext_resource type=\"Texture2D\" path=\"%1\" id=\"%2\"]\n")
+            .arg(sanitizeQuotedString(it.key()), sanitizeQuotedString(it->id)).toUtf8());
+    }
+    device->write("\n");
+
+    // TileSetAtlasSource nodes
+    for (auto itTileset = assetInfo.tilesetInfo.begin(); itTileset != assetInfo.tilesetInfo.end(); ++itTileset) {
+        if (itTileset->usedTiles.size() == 0)
+            continue;
+
+        device->write(QString("[sub_resource type=\"TileSetAtlasSource\" id=\"TileSetAtlasSource_%1\"]\n")
+            .arg(QString::number(itTileset->atlasId)).toUtf8());
+
+        device->write(QString("texture = ExtResource(\"%1\")\n")
+            .arg(sanitizeQuotedString(itTileset->id)).toUtf8());
+
+        if (itTileset->tileset->margin() != 0)
+            device->write(QString("margins = Vector2i(%1, %1)\n")
+                .arg(QString::number(itTileset->tileset->margin())).toUtf8());
```

Comment:
Let's use the overload of `arg` that takes a number for brevity:
```suggestion
                .arg(itTileset->tileset->margin());
```

Reply:
This won't work on versions of Qt older than 5.14 (5.12 is common, since it's the version the offline installer is pegged at), but I made some convenience functions to get the same effect. Thanks for the push :)